File: .repo/system-design-mastery-module-2-database-fundamentals/2-replication-and-partitioning/2-csharp/Controllers/OrdersController.cs
using Microsoft.AspNetCore.Mvc;
using ReplicationPartitioning.Models;
using ReplicationPartitioning.Services;
using System.Threading.Tasks;

namespace ReplicationPartitioning.Controllers
{
    [ApiController]
    [Route("api")]
    public class OrdersController : ControllerBase
    {
        private readonly OrdersService _ordersService;

        public OrdersController(OrdersService ordersService)
        {
            _ordersService = ordersService;
        }

        [HttpPost("seed/{count}")]
        public async Task<IActionResult> Seed([FromRoute] int count)
        {
            var result = await _ordersService.SeedAsync(count);
            return StatusCode(201, result);
        }

        [HttpPost("orders")]
        public async Task<IActionResult> Create([FromBody] OrderEntity order)
        {
            var created = await _ordersService.CreateAsync(order);
            return Created($"/api/orders/{created.Id}", created);
        }

        [HttpGet("orders")]
        public async Task<IActionResult> FindAll()
        {
            var orders = await _ordersService.FindAllAsync();
            return Ok(orders);
        }

        [HttpGet("replication/status")]
        public async Task<IActionResult> GetReplicationStatus()
        {
            var status = await _ordersService.GetReplicationStatusAsync();
            return Ok(status);
        }

        [HttpGet("partitions")]
        public async Task<IActionResult> GetPartitionInfo()
        {
            var partitions = await _ordersService.GetPartitionInfoAsync();
            return Ok(partitions);
        }
    }
}

File: .repo/system-design-mastery-module-2-database-fundamentals/2-replication-and-partitioning/1-java/src/main/java/academy/starci/replication/orders/OrdersController.java
package academy.starci.replication.orders;

import org.springframework.http.HttpStatus;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.*;
import java.util.List;
import java.util.Map;

@RestController
@RequestMapping("/api")
public class OrdersController {

    private final OrdersService ordersService;

    public OrdersController(OrdersService ordersService) {
        this.ordersService = ordersService;
    }

    @PostMapping("/seed/{count}")
    public ResponseEntity<Map<String, Object>> seed(@PathVariable("count") int count) {
        Map<String, Object> result = ordersService.seed(count);
        return ResponseEntity.status(HttpStatus.CREATED).body(result);
    }

    @PostMapping("/orders")
    public ResponseEntity<OrderEntity> create(@RequestBody OrderEntity order) {
        OrderEntity created = ordersService.create(order);
        return ResponseEntity.status(HttpStatus.CREATED).body(created);
    }

    @GetMapping("/orders")
    public ResponseEntity<List<OrderEntity>> findAll() {
        List<OrderEntity> result = ordersService.findAll();
        return ResponseEntity.ok(result);
    }

    @GetMapping("/replication/status")
    public ResponseEntity<List<Map<String, Object>>> getReplicationStatus() {
        List<Map<String, Object>> result = ordersService.getReplicationStatus();
        return ResponseEntity.ok(result);
    }

    @GetMapping("/partitions")
    public ResponseEntity<List<Map<String, Object>>> getPartitionInfo() {
        List<Map<String, Object>> result = ordersService.getPartitionInfo();
        return ResponseEntity.ok(result);
    }
}
